app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    !current_user.nil?
+  end
+
+  # Logs out the current user.
+  def log_out
+    session.delete(:user_id)
+    @current_user = nil
+  end
```

Comment:
1. Xóa mấy cái comment đi nhé. Code clear rồi mà
2. giải thích giúp chị `current_user` có dấu thang (`!`) là có ý nghĩa gì nhé

Reply:
Nếu  chưa có chưa có @current_user = nil thì @current_user.nil? = true dẫn đến sai sót nên phải dùng thêm toán tử( ! )

Reply:
rứa phân biệt giúp chị 3 method sau:
 nil?
present?
any?
Sau đó cân nhắc xem method nào sẽ phù hợp với mục đích sử dụng của em mà ko cần dùng toán tử đảo ngược(!)

Reply:
nil? trả về true hay flase nếu biến đó là nil hay không nil

Reply:
Khóc. Tìm hiểu giúp chị 3 method sau: nil?, present?, any?
Và cái giá trị của
[!current_user.nil?] giống kết quả với [current_user.present?] á
Nên dùng presents?  nhé
:P